Anature/src/application/controllers/BattleController.java

```diff
@@ -372,6 +373,67 @@ private void setUpDialogue(Scene scene)
 		mDialogueTxtArea.fontProperty().bind(fontProperty);
 	}
 
+	private void onPlayerAnatureDeath()
+	{
+		boolean isThereAliveAnatureInParty = false;
+		for(Anature anature : mPlayer.getAnatures())
+		{
+			if(anature.getCurrHp() > 0)
+			{
+				isThereAliveAnatureInParty = true;
+			}
+		}
+
+		if(isThereAliveAnatureInParty)
+		{
+			onSwitchBtn();
+			mShowSwitchBackBtn.set(false);
+		}
+		else
+		{
+			mDialogueTxt.set(mFightManager.getPlayerTeam().get(0).getName() + " has been defeated!");
+
+			mShowBtns.set(false);
+
+			mClickQueue.clear();
+			mClickQueue.enqueue(() -> mDialogueTxt.set("You have no more Anatures! You quickly run back to the nearest Rest Station!"));
+			mClickQueue.enqueue(() -> Startup.changeScene(SceneType.Starter_Town));
```

Comment:
This needs to pass in (null, null) that way the model knows to switch to whatever scene the player was in previously.